src/v2/view-builder/views/captcha/CaptchaView.js

```diff
@@ -49,22 +74,58 @@ export default View.extend({
     // Callback invoked when CAPTCHA is solved. We're binding it to this view so that it's easier
     // to unit test.
     this.onCaptchaSolved = (token) => {
+      const captchaObject = this._getCaptchaOject();
+
+      // We reset the Captcha. We need to reset because every time the 
+      // Captcha resolves back with a token and say we have a server side error, 
+      // if we submit the form again it won't work otherwise. The Captcha 
+      // has to be reset for it to work again in that scenario.
+      const captchaId = this.$el.find('#captcha-container').attr('data-captcha-id');
+      captchaObject.reset(captchaId);
 
       // Set the token in the model
       const fieldName = this.options.name;
       this.model.set(fieldName, token);
 
-      // Clear form errors before re-validation
-      this.model.trigger('clearFormError');
+      // By the time we reach this callback, the model has already unflattened the field names.
```

Comment:
Doesn't seem right.
`this.model` shall always stay flatten. `this.model.toJSON` would return unflattened data.

Reply:
It's fixed at https://github.com/okta/okta-signin-widget/pull/1873/files

Reply:
The issue that I found is that `this.model.toJSON` is invoked before this `onCaptchaLoaded` is invoked in the case where there is validation errors (because the view has to re-render itself to display client side errors). Because of that, by the time it gets to this callback, the model itself contains both the flattened and unflattened data - so it essentially has two copies of the underlying. I'm not sure if this is an intended behavior or a bug but here's a little demo of this:

https://user-images.githubusercontent.com/77295104/117023965-939de500-acc7-11eb-9757-bf778fe41722.mov

Reply:
UDPATE: The issue I demoed above is caused by the underlying problem that this PR is solving: https://github.com/okta/okta-signin-widget/pull/1873/files. Thank you @haishengwu-okta for pointing this out. All this extra logic around unflattening is not needed and I've removed it now.